.github/workflows/main.yml

```diff
@@ -0,0 +1,44 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches:
+      - '*'
+    tags:
+      - '*'
+  pull_request:
+    branches:
+      - 'main'
```

Comment:
i would suggest `branches: [ '*' ]`  because we want to run the `tests` job when there is a PR against any branch.